sw/device/tests/sim_dv/spi_tx_rx_test.c

```diff
@@ -0,0 +1,391 @@
+                                  0x1) == kDifPlicOk,
+        "dif_plic_irq_priority_set failed");
+  CHECK(dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxlvl,
+                                  0x1) == kDifPlicOk,
+        "dif_plic_irq_priority_set failed");
+  CHECK(dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceTxlvl,
+                                  0x1) == kDifPlicOk,
+        , "dif_plic_irq_priority_set failed");
+  CHECK(dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxerr,
+                                  0x1) == kDifPlicOk,
+        "dif_plic_irq_priority_set failed");
+  CHECK(dif_plic_irq_priority_set(
+            plic, kTopEarlgreyPlicIrqIdSpiDeviceRxoverflow, 0x2) == kDifPlicOk,
+        "dif_plic_irq_priority_set failed");
+  CHECK(dif_plic_irq_priority_set(
+            plic, kTopEarlgreyPlicIrqIdSpiDeviceTxunderflow, 0x2) == kDifPlicOk,
+        "dif_plic_irq_priority_set failed");
+
+  // Set the threshold for the Ibex to 0.
+  CHECK(dif_plic_target_threshold_set(plic, kTopEarlgreyPlicTargetIbex0, 0x0) ==
+            kDifPlicOk,
+        "dif_plic_target_threshold_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxf,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxlvl,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceTxlvl,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxerr,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceRxoverflow,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdSpiDeviceTxunderflow,
+                                kTopEarlgreyPlicTargetIbex0,
+                                kDifPlicEnable) == kDifPlicOk,
+        "dif_plic_irq_enable_set failed");
+}
+
+static bool execute_test(const dif_spi_device_t *spi_device) {
+  size_t bytes_recved = 0;
+  size_t bytes_transferred = 0;
+  uint8_t spi_device_rx_data[SPI_DEVICE_DATASET_SIZE];
+  uint16_t rx_level = SPI_DEVICE_DATASET_SIZE;
+  uint16_t tx_level = SPI_DEVICE_DATASET_SIZE / 2;
```

Comment:
Do not pre-declare variables. Instead, move these down to where they are first used.